src/main/java/seedu/address/model/assignment/DueDate.java

```diff
@@ -2,28 +2,58 @@
 
 import static java.util.Objects.requireNonNull;
 
+import java.time.LocalDate;
 import java.time.LocalDateTime;
+import java.time.LocalTime;
 import java.time.format.DateTimeFormatter;
 
 public class DueDate {
     public static final String MESSAGE_CONSTRAINTS = "Due dates should be in a format dd/MM/yyyy HHmm";
-    //TODO: decide which parse, output format to keep
-    protected static final DateTimeFormatter PARSE_FORMAT = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+
+    protected static final DateTimeFormatter PARSE_DATE_FORMAT = DateTimeFormatter.ofPattern("d/M/yyyy");
+    protected static final DateTimeFormatter PARSE_TIME_FORMAT = DateTimeFormatter.ofPattern("HHmm");
     protected static final DateTimeFormatter OUTPUT_FORMAT = DateTimeFormatter.ofPattern("dd MMM yyyy, hh:mm a");
+
     //TODO: change VALIDATION INDEX
     //public static final String VALIDATION_REGEX = "\\d{3,}";
-    public final LocalDateTime value;
+
+    private final LocalDate date;
+    private final LocalDateTime dateTime;
+    private final LocalTime time;
+
+    public final String value;
 
     /**
      * Constructs a {@code DueDate}.
      *
-     * @param dateTime A valid dateTime.
+     * @param date Date of due date.
+     * @param time Time of due date.
      */
-    public DueDate(LocalDateTime dateTime) {
-        requireNonNull(dateTime);
+    public DueDate(String date, String time) {
+        requireNonNull(date);
+        requireNonNull(time);
         // TODO: use Validation regex to verify dateTime format
         //checkArgument(isValidPhone(phone), MESSAGE_CONSTRAINTS);
-        this.value = dateTime;
+        this.date = LocalDate.parse(date, PARSE_DATE_FORMAT);
+        this.time = LocalTime.parse(time, PARSE_TIME_FORMAT);
+        this.dateTime = LocalDateTime.of(this.date, this.time);
+        this.value = this.dateTime.format(OUTPUT_FORMAT);
+    }
+
+    /**
+     * Constructs a {@code DueDate}.
+     *
+     * @param date Date of due date.
+     */
+    public DueDate(String date) {
```

Comment:
I thought about the overloaded constructor yesterday. How about we do this instead?

```
public DueDate(String date) {
    new DueDate(date, "2359");
}
```

Reply:
I think that is better, but when I do that, the class fields will be highlighted in red and says might not have been intialised.

Reply:
![image](https://user-images.githubusercontent.com/77260240/134880000-0f3e6c72-1955-4db7-8051-88d78513cb95.png)

Reply:
Looks like a tricky problem. To use this method, we may need some default values for all the fields in the class, which may be harder to understand. I guess we will go with your method!

Reply:
this(date, "2359"); TRY this, this may work,